saleor/graphql/order/mutations/draft_order_update.py

```diff
@@ -336,28 +356,51 @@ def perform_mutation(cls, _root, info: ResolveInfo, /, **data):
         private_metadata_collection = cls.create_metadata_from_graphql_input(
             private_metadata_list, error_field_name="private_metadata"
         )
-
-        instance = cls.construct_instance(instance, cleaned_input)
-
         cls.validate_and_update_metadata(
             instance, metadata_collection, private_metadata_collection
         )
-        cls.clean_instance(info, instance)
-        new_instance_data = instance.serialize_for_comparison()
-        changed_fields = cls.diff_instance_data_fields(
-            instance.comparison_fields,
-            old_instance_data,
-            new_instance_data,
+
+    @classmethod
+    def perform_mutation(cls, _root, info: ResolveInfo, /, **data):
+        instance = cls.get_instance(info, **data)
+        channel_id = instance.channel_id
+        cls.check_channel_permissions(info, [channel_id])
+
+        instance = cast(models.Order, instance)
+        instance_tracker = InstanceTracker(instance, cls.FIELDS_TO_TRACK)
+        shipping_address_tracker = InstanceTracker(
+            instance.shipping_address, ADDRESS_UPDATE_FIELDS
         )
-        cls._save(
-            info, instance, cleaned_input, old_voucher, old_voucher_code, changed_fields
+        billing_address_tracker = InstanceTracker(
+            instance.billing_address, ADDRESS_UPDATE_FIELDS
         )
+
+        old_voucher = instance.voucher
+        old_voucher_code = instance.voucher_code
+        data = data["input"]
+        cleaned_input = cls.clean_input(info, instance, data)
+
+        cls.handle_metadata(instance, cleaned_input)
+
+        # update instance with data from input
+        instance = cls.construct_instance(instance, cleaned_input)
+
+        cls.clean_instance(info, instance)
+
+        manager = get_plugin_manager_promise(info.context).get()
+        cls.handle_shipping(cleaned_input, instance, manager)
+        cls.handle_order_voucher(cleaned_input, instance, old_voucher, old_voucher_code)
+
+        order_modified = cls._save(
+            instance_tracker,
+            shipping_address_tracker,
+            billing_address_tracker,
+            cleaned_input,
+        )
+
+        if order_modified:
+            cls._post_save_action(instance, manager)
+
         cls._save_m2m(info, instance, cleaned_input)
```

Comment:
Hhm, what about this? 🤔  It looks like previously we were trigerring the webhook before calling save on m2m. Do we still need it? If yes, the probably we should change the order of webhook calls.

Reply:
The order is the same: we trigger the webhook before save m2m

Reply:
OK, got your point. Let me correct this